cloud/cluster/cluster.go

```diff
@@ -0,0 +1,64 @@
+package cluster
+
+import (
+	"fmt"
+	"io"
+)
+
+type NodeId string
+
+type Node interface {
+	Id() NodeId
```

Comment:
Do we want to expand this to something more definitive than an Id?  Like host, port, etc... so the functions that take in a Node and generate a worker don't have to parse some special format?

At the very least we need a comment specifying how to get the connection info from the Id()